keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepositoryTests.kt

```diff
@@ -236,6 +237,25 @@ abstract class ResourceRepositoryTests<T : ResourceRepository> : JUnit5Minutests
                 }
             }
           }
+
+          context("appending an identical event with duplicates disallowed, but after an application event") {
+            before {
+              tick()
+              subject.appendHistory(ResourceValid(resource, clock))
+              tick()
+              subject.appendHistory(ApplicationActuationResumed("fnord", "keel@keel.io"))
+              tick()
+              subject.appendHistory(ResourceValid(resource, clock))
+            }
+
+            test("overrides the duplicate check and persists the event") {
+              expectThat(subject.eventHistory(resource.id))
+                .and {
+                  first().isA<ResourceValid>()
+                  second().isA<ResourceValid>()
```

Comment:
This is failing here because the second event is never inserted into the DB, judging from the jooq debug logs, and yet when I run it in the debugger I can follow it to the insert path... 🤔